core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java

```diff
@@ -192,7 +192,7 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                                 fail(ctx, ContentTooLargeException.get());
                                 return;
                             } else {
-                                res.write(HttpData.of(data));
+                                res.write(HttpData.copyOf(data));
```

Comment:
I'm curious if we can use `wrap()` here.

Reply:
Yeah noticed this - didn't want to think too much about it but changed to `wrap(retain)` since seems ok.